Shake/Main.lean

```diff
@@ -458,14 +458,19 @@ def main (args : List String) : IO UInt32 := do
   else pure none
 
   -- Read the config file
-  let cfg ← if let some file := cfgFile then
+  -- Set `validCfgFile?` to `false` only if the config file is present and invalid
+  let (cfg, validCfgFile?) ← if let some file := cfgFile then
     try
-      IO.ofExcept (Json.parse (← IO.FS.readFile file) >>= fromJson? (α := ShakeCfg))
+      pure (← IO.ofExcept (Json.parse (← IO.FS.readFile file) >>= fromJson? (α := ShakeCfg)), true)
     catch e =>
+      -- the `cfgFile` is invalid, so we print the error and return `validCfgFile? = false`
       println! "{e.toString}"
-      pure {}
-  else pure {}
-
+      pure ({}, false)
+  else pure ({}, true)
+  if ! validCfgFile? then
```

Comment:
How about inlining this into the only branch setting the option to `false`?

Reply:
I am not able to test now, but can you actually do that?  I think that the "`false`" branch is inside a `try ... catch` inside a `let` and some of that may prevent exiting early the program.

Reply:
Oh, you're right. Inlining this branch locally made shake barf violently - so your current suggestion may be best.
One style nit, though:
```suggestion
if !validCfgFile? then

Reply:
Or even "isValidCfgFile", as this returns a bool and not an option. What do you think?

Reply:
I used `is` and revised the whitespace, thanks!